qemu/tests/msi_change_flag.py

```diff
@@ -0,0 +1,91 @@
+    """
+    tmp_dir = params["tmp_dir"]
+    filesize = int(params.get("filesize"))
+    dd_cmd = params["dd_cmd"]
+    file_md5_check_timeout = int(params.get("file_md5_check_timeout"))
+
+    def get_file_md5sum(file_name, session, timeout):
+        """
+        Get file md5sum from guest.
+        """
+        logging.info("Get md5sum of the file:'%s'" % file_name)
+        s, o = session.cmd_status_output("md5sum %s" % file_name, timeout=timeout)
+        if s != 0:
+            test.error("Get file md5sum failed as %s" % o)
+        return re.findall(r"\w{32}", o)[0]
+
+    login_timeout = int(params.get("login_timeout", 360))
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_serial_login()
+
+    if params.get("os_type") == "windows":
+        driver = params.get("driver_name")
+        error_context.context("Boot guest with %s device" % driver, logging.info)
+        session = utils_test.qemu.windrv_check_running_verifier(session, vm,
+                                                                test, driver,
+                                                                login_timeout)
+        error_context.context("Check %s's irq number" % params.get("device_name"), logging.info)
+        devcon_folder = utils_misc.set_winutils_letter(session, params["devcon_folder"])
+        if params.get("msi_cmd"):
+            error_context.context("Disable MSI in guest", logging.info)
+            hwid = win_dev.get_hwids(session, params.get("device_name"), devcon_folder, login_timeout)[0]
+            session.cmd(params["msi_cmd"] % (hwid, 0))
+            session = vm.reboot(session=session)
+
+    # prepare test data
+    guest_path = (tmp_dir + "src-%s" % utils_misc.generate_random_string(8))
+    host_path = os.path.join(test.tmpdir, "tmp-%s" %
+                             utils_misc.generate_random_string(8))
+    logging.info("Test setup: Creating %dMB file on host", filesize)
+    process.run(dd_cmd % (host_path, filesize), shell=True)
+
+    try:
+        src_md5 = (crypto.hash_file(host_path, algorithm="md5"))
+        logging.info("md5 value of data from src: %s" % src_md5,
+                              logging.info)
+        # transfer data
+        error_context.context("Transfer data from host to %s" % vm.name,
+                              logging.info)
+        vm.copy_files_to(host_path, guest_path)
+        dst_md5 = get_file_md5sum(guest_path, session,
+                                  timeout=file_md5_check_timeout)
+        logging.info("md5 value of data in %s: %s" % (vm.name, dst_md5),
+                              logging.info)
+        if dst_md5 != src_md5:
+            test.fail("File changed after transfer host -> %s" % vm.name)
+        else:
+            logging.info("md5 values of host and %s are equal" % vm.name)
+    finally:
+        os.remove(host_path)
+        if params.get("os_type") == "linux":
+            session.cmd("rm -rf %s" % guest_path,
```

Comment:
I think can define a delete command in cfg, and then you don't need to judge the guest here.

Reply:
But the paths of Windows and rhel are different,if define a delete command in cfg,when using it still need to judge the guest.

Reply:
I saw line4 " #In this test need set snapshot for our test will chang guest msi support", so need we still clean the env. in guest?

Reply:
> But the paths of Windows and rhel are different,if define a delete command in cfg,when using it still need to judge the guest.

Like `session.cmd(delete_cmd % guest_path)`